compiler_gym/leaderboard/llvm_instcount.py

```diff
@@ -253,33 +264,50 @@ def main(argv):
             # If we are resuming from a previous job, read the states that have
             # already been proccessed and remove those benchmarks from the list
             # of benchmarks to evaluate.
-            init_states = []
+            states = []
             if FLAGS.resume and Path(FLAGS.leaderboard_results).is_file():
                 with CompilerEnvStateReader(open(FLAGS.leaderboard_results)) as reader:
                     for state in reader:
-                        init_states.append(state)
+                        states.append(state)
                         if state.benchmark in benchmarks:
                             benchmarks.remove(state.benchmark)
 
             # Run the benchmark loop in background so that we can asynchronously
             # log progress.
-            worker = _EvalPolicyWorker(env, benchmarks, policy, init_states)
+            # split the tasks into multiple processes.
+            q = mp.Queue()
+            worker = _EvalPolicyWorker(num_process, benchmarks, policy, q)
             worker.start()
+
             timer = Timer().reset()
             try:
+                # Determine if we need to print a header.
+                header = (
+                    not Path(FLAGS.leaderboard_results).is_file()
+                    or os.stat(FLAGS.leaderboard_results).st_size == 0
+                )
+                writer = CompilerEnvStateWriter(open(FLAGS.leaderboard_results, "a"), header=header)
                 print(
                     f"=== Evaluating policy on "
                     f"{humanize.intcomma(total_count)} "
                     f"{FLAGS.test_dataset} benchmarks ==="
                     "\n\n"  # Blank lines will be filled below
                 )
-                while worker.is_alive():
-                    done_count = len(worker.states)
+
+                while len(states) < len(benchmarks):
```

Comment:
There might be a bug here. If you check the line `if FLAGS.resume and Path(FLAGS.leaderboard_results).is_file():` above, the `len(benchamrks)` might be less than the `len(states)` when resuming from an incomplete run (not entirely sure, would need to test :) )